Remove `pkg_resources`

ndmlny-qs opened the pull request:
This commit removes `pkg_resources` requirements as they are deprecated with more recent versions of Python.

- `importlib.metadata.distribution` is used in place of `pkg_resources.get_distribution`
- `importlib.metadata.version` is used in place of `pkg_resources.version`
- `ModuleNotFoundError` is used instead of `pkg_resources.DistributionNotFound`

## Todos

- [ ] MIT compatible
- [ ] Tests
- [ ] Documentation
- [ ] Updated CHANGES.rst

ndmlny-qs committed:
Remove `pkg_resources`

This commit removes `pkg_resources` requirements as they are deprecated
with more recent versions of Python.

- `importlib.metadata.distribution` is used in place of
  `pkg_resources.get_distribution`
- `importlib.metadata.version` is used in place of
  `pkg_resources.version`
- `ModuleNotFoundError` is used instead of
  `pkg_resources.DistributionNotFound`

mlucool commented on sqlalchemy_redshift/__init__.py:
```diff
@@ -1,14 +1,19 @@
+from importlib.metadata import version
+
+from packaging.version import parse as parse_version
+from pkg_resources import get_distribution
```
`pkg_resources` is still imported and used for `sqlalchemy_redshift`. Can it be totally removed?

ndmlny-qs replied:
done

ndmlny-qs committed:
Remove all traces of `pkg_resources`

This commit removes all references to `pkg_resources`.

- Bumped version numbers for found in `requirements-docs` in order to
  build the documentation using `tox`.
- Removed all references to `pkg_resources` in
  `sqlalchemy_redshift/__init__.py` and
  `sqlalchemy_redshift/dialect.py`.
- Updated the `tests/test_default_ssl.py` to no longer use
  `pkg_resources`.

ndmlny-qs commented:
@Brooke-white is there someone I should ping to look at this PR, or is this also meant to be a part of the work being done in #264?

Thanks again for the tool, yours and everyone else's work with it has been greatly appreciated.

peytondmurray commented:
Hey all, I'm interested in getting this merged. Is there anything I can do to help get this across the line?

jaellis commented:
Bump

TeddyCr commented:
Thank you for your contribution and interest in sqlalchemy-redshift. With the help of the OpenMetadata engineering team I recently released sqlalchemy-redshift 1.0.0 with support for SQLAlchemy 2.

With this recent change I'll be closing open tickets and PRs. If you find your issue or PR still relevant for sqlalchemy-redshift v1.0.0, please reopen the ticket.

You can find more information here https://medium.com/p/1388d8dec83d
